### PR TITLE
Add warnings for sorting and comparison

### PR DESCRIPTION
**Pull request by nanjekyejoannah**
Most of the warnings are covered on the list sort method.

I added the missing warnings for the `cmp` and `__cmp__` method.

This PR builds on #3 hence the other changes are showing because I had to use the new exception.

### CODE REVIEW

**Comment by nanjekyejoannah** [comment]
For some reason I cant assign people to PRs  on this project. I dont know if its a permissions thing.

**Comment by vext01** [comment]
Before I prod around in the organisation settings, do you see this box when viewing the PR page?

image

**Comment by nanjekyejoannah** [comment]
No, I cant see the above on this repository, I confirmed that on the CPython repositoryI can, thats why I thought its a rights thing.

Sorry been out for meetings back to back.

**Comment by vext01** [comment]
Can you see the assign box now?

**Comment by nanjekyejoannah** [comment]
Yes, I can now assign, ad labels etc, thank you.

File: Objects/exceptions.c [hunk 1]
```diff
@@ -1981,6 +1981,13 @@ SimpleExtendsException(PyExc_Warning, SyntaxWarning,
                        "Base class for warnings about dubious syntax.");
 
 
+/*
+ *    3xWarning extends Warning
+ */
+SimpleExtendsException(PyExc_Warning, Py3xWarning,
```
Comment (ltratt) [inline Objects/exceptions.c:1988]: It feels like we probably want to break this exception class out into its own commit if possible (and then we can explain the commit message why we've done it, what the implications are, etc).

File: Lib/test/test_grammar.py [hunk 2]
```diff
@@ -65,6 +65,15 @@ def test_plain_integers(self):
         else:
             self.fail('Weird maxint value %r' % maxint)
 
+        if sys.py3kwarning:
+            with warnings.catch_warnings():
+                warnings.filterwarnings('error', category=Py3xWarning)
+                with self.assertRaises(Py3xWarning) as oct:
+                    compile('032', '<test string>', 'eval')
+                self.assertIn("octal literals are not supported in 3.x;\n" 
+                              "drop the leading 0",
```
Comment (ltratt) [inline Lib/test/test_grammar.py]: I don't think that dropping the leading `0` is the right fix: I *think* that the user has to translate their octal literal into another base entirely. This is an instance where customising the fix to to the user's input might be helpful (e.g. we might say "change 0o30 to 24").

File: Objects/object.c [hunk 3]
```diff
@@ -819,6 +819,10 @@ do_cmp(PyObject *v, PyObject *w)
 {
     int c;
     cmpfunc f;
+    if (Py_Py3kWarningFlag &&
```
Comment (ltratt) [inline Objects/object.c]: I wonder if sometimes we can give a more precise fix to the user? I'm not sure it's possible in general though.

File: Objects/exceptions.c (hunk 1 above)
Comment (nanjekyejoannah) [reply Objects/exceptions.c:1988]: The associated PR might have been merged but for some reason the diff still shows even after merging wierd. We can see what do with @vext01

Comment (vext01) [reply Objects/exceptions.c:1988]: Did you rebase?

Comment (nanjekyejoannah) [reply Objects/exceptions.c:1988]: I might have just started a new branch off the numbers one, then opened the PR, since it was opened before the numbers PR was merged

Comment (ltratt) [reply Objects/exceptions.c:1988]: If @vext01 is OK with you doing so, it might be possible (and worthwhile) to do a rebase now and see if that sorts things out. I'll leave that to the two of you!

Comment (vext01) [reply Objects/exceptions.c:1988]: I'm OK with that.

**Comment by vext01** [comment]
where do we stand with this PR?
